File: Oblig1/Rack.java
/*
Klassen Rack.

Held opp til eit gjeven tal med nodar i ei ArrayList.
Klassa hentar i hovudsak informasjon fraa nodane objektet held.
*/

//For aa kunne oppretta ei ArrayList
import java.util.ArrayList;

public class Rack
{
  //Oppretter instansvariablane:
  //Maks tal paa nodar i racket
  int maksAntNoder;
  //Liste som held nodeobjekt
  ArrayList<Node> noder;

  //Konstruktoeren,
  //som tek inn makstalet paa nodarr og sett verdien til instansvariabelen,
  //og gjev verdi til lista m/nodar
  public Rack(int maksAntNoderInn)
  {
    maksAntNoder = maksAntNoderInn;
    noder = new ArrayList<Node>();
  }

  //Funksjon som sjekker om det er leidig plass i racket,
  //returnerar sanningsverdi
  public boolean sjekkLedigPlass()
  {
    //Sett ikkje leidig plass
    boolean ledigPlass = false;
    //Viss makstal for nodar i rack er stoerre enn tal paa element i ArrayList...
    if(maksAntNoder > noder.size())
    {
      //er det leidig plass.
      ledigPlass = true;
    }
    //Returner om det er leidig plass eller ikkje
    return ledigPlass;
  }

  //Metode som legg til nytt nodeobjekt i ArrayList
  public void leggTilNode(Node nyNode)
  {
    noder.add(nyNode);
  }

  //Funksjon som les talet paa prosessorar i racket,
  //returnerar heiltalsverdi
  public int lesAntProsessorer()
  {
    int prosessorer = 0;
    //For kvar node i racket (i ArrayList)...
    for(Node i : noder)
    {
      //Legg til talet paa prosessorar.
      prosessorer += i.lesAntProsessorer();
    }
    //Returner talet paa prosessorar
    return prosessorer;
  }

  //Funksjon som les talet paa nodar i racket,
  //returnerar heiltalsverdi
  public int lesAntNoder()
  {
    //Returnerar talet paa element i ArrayList
    return noder.size();
  }

//Funksjon som sjekker om kvar node i racket har eit minimum tal m/minne,
//returnerar heiltalsverdi
  public int noderMedNokMinne(int paakrevdMinne)
  {
    //Teljar for godkjente nodar
    int noderGodkjente = 0;
    //For kvar node i racket (i ArrayList)...
    for(Node i : noder)
    {
      //Viss noden har meir eller likt minimumskravet for minne...
      if(i.lesMinneGB() >= paakrevdMinne)
      {
        //legg ein til teljaren
        noderGodkjente++;
      }
    }
    //Returnerar talet paa godkjente nodar
    return noderGodkjente;
  }
}

File: Oblig1/python/rack.py
"""
Held klassen Rack.

Held opp til eit gjeve tal nodar i ei liste.
Klassa hentar i hovudsak informasjon fraa nodane ho held.
"""

class Rack:
    #Konstruktoeren, tek inn makstalet paa noder instansen kan helda
    def __init__(self, maksAntNoder):
        #sett minnekapasiteten til det gjevne talet
        self._maksAntNoder = maksAntNoder
        #Lista som held nodane i racket
        self._noder = []

    #Funksjon som sjekker om det er ledig plass i racket,
    #Returnerar ein sanningsverdi
    def sjekkLedigPlass(self):
        #sett ikkje ledig plass
        ledigPlass = False
        #Viss minnekapasiteten er stoerre enn talet paa nodar...
        if self._maksAntNoder > len(self._noder):
            #Er det ledig plass i racket
            ledigPlass = True
        #Returner om det er ledig plass eller ikkje
        return ledigPlass

    #Metode som legg til node i dette racket
    def leggTilNode(self, nyNode):
        self._noder.append(nyNode)

    #Funksjon som returnerar talet paa prosessorar i racket
    def antProsessorer(self):
        prosessorer = 0
        #For kvar node i racket...
        for i in self._noder:
            #Legg til talet paa prosessorar
            prosessorer += i.antProsessorer()
        #Returner talet
        return prosessorer

    #Returnerar talet paa nodar i racket
    def antNoder(self):
        #Returner talet paa element i lista _noder
        return len(self._noder)

    #Funksjon som tek inn paakrevd minnestorleik som parameter,
    #Og returnerar talet paa nodar som har god nok kapasitet
    def noderMedNokMinne(self, paakrevdMinne):
        noder = 0
        #For kvar node i racket...
        for i in self._noder:
            #Viss minnestorleiken i noden er stoerre eller erlik paakrevd minnestorleik...
            if i.minneIGB() >= paakrevdMinne:
                #Legg til eit tal i teljaren
                noder += 1
        #Returner talet
        return noder
